pytorch_to_returnn/converter/converter.py

```diff
@@ -156,6 +157,27 @@ def _make_tf_feed_dict(self, input: Data):
     else:
       for i, size in input.size_placeholder.items():
         d[size] = [self._inputs_np.shape[input.get_batch_axis(i)]] * n_batch  # not so relevant
+    if out_entry:
+      visited_entries = set()
+
+      def _get_non_deterministic_tensor_entries(entry_: TensorEntry):
+        """
+        Go recursively through all tensor entries to find non deterministic ones
+        """
+        if not all(call_.module.module.is_deterministic for call_ in entry_.output_from_calls):
+          return [entry_]
+        non_det_mods = []
+        for call_ in entry_.output_from_calls:
+          for prev_entry_ in [pe for pe in call_.inputs_args if isinstance(pe, TensorEntry)]:
+            if prev_entry_ != entry_ and prev_entry_ not in visited_entries:
+              visited_entries.add(prev_entry_)
```

Comment:
I would move that `visited_entries` logic above (see other comment).
And also, I don't think you need the `prev_entry_ != entry_` check anymore then.